utils/db_process.py

```diff
@@ -0,0 +1,93 @@
+import mysql.connector
+from mysql.connector import pooling
+import os
+from dotenv import load_dotenv
+from typing import Union, Tuple, Dict, List
+from typing import Final
+import datetime
+load_dotenv()
+
+pool_config = {
+    "pool_name": "pool",
+    "pool_size": 5,
+    "host": os.getenv("DB_HOST"),
+    "user": os.getenv("DB_USER"),
+    "password": os.getenv("DB_PASSWORD"),
+    "database": os.getenv("DB_DATABASE"),
+}
+pool_config: Final
+
+
+cnx_pool = pooling.MySQLConnectionPool(**pool_config)
+
+
+def get_all_result(sql, param: Union[Tuple, None] = None) -> Dict:
+    try:
+        connection = cnx_pool.get_connection()
+        cursor = connection.cursor()
+
+        if param is None:
+            cursor.execute(sql)
+        else:
+            cursor.execute(sql, param)
+        result = cursor.fetchall()
+        
+        cursor.close()
+        connection.close()
+        column_names = [desc[0] for desc in cursor.description]
+        result = [dict(zip(column_names, row)) for row in result]
+        return result
+
+    except mysql.connector.Error as e:
+        print(e)
+        cursor.close()
+        connection.close()
```

Comment:
Vars can be not declared when there's something wrong to the connection. Tbh, I don't really know how Python's errors are handled(or the life scope thing).